File: js/axios/api-keys/fetchKeys.js
import axios from "axios";
import "dotenv/config";

const accessKey = process.env.ACCESS_TOKEN;
const apiUrl = process.env.production
  ? "https://api.theauthapi.com"
  : process.env.TESTING_URL;
const projectId = process.env.PROJECT_ID;

async function fetchApiKeys(filters) {
  try {
    const {data} = await axios
      .get(`${apiUrl}/api-keys${filters ? `?${filters}` : ''}`, {
        headers: {
          ContentType: "application/json",
          "x-api-key": accessKey,
        },
      });
    return data;
  } catch (error) {
    if (error.response) {
      console.log(error.response.data);
    } else {
      // handle other errors
    }
  }
}

function getFiltersQuery(filters) {
  return Object.entries(filters).map(([key, value]) => `${key}=${value}`)
      .join('&');
}

const filters = "";
//const filters = "?name=sdfsdfsdfsdf&isActive=false";
//const filters = "?isActive=false";
//const filters = "?customAccountId=VALUE";
//const filters = "?customUserId=VALUE";

(async () => {
  // fetch all keys
  const apiKeys = await fetchApiKeys(filters);
  console.log('All Keys', apiKeys);

  // get keys with a specific name
  const nameFilteredKeys = await fetchApiKeys(getFiltersQuery({
    name: "new api key",
  }));
  console.log('Keys filtered using name', nameFilteredKeys);

  // fetch keys where customUserId is null
  const customUserIdFilteredKey = await fetchApiKeys(getFiltersQuery({
    customUserId: null,
  }));
  console.log('Keys filtered using customUserId', customUserIdFilteredKey);

  // fetch inactive (revoked) keys
  const inactiveKeys = await fetchApiKeys(getFiltersQuery({
    isActive: false,
  }));
  console.log('Inactive (revoked) keys', inactiveKeys);
})();

File: python/api-keys/fetch_keys.py
import requests
from requests.exceptions import HTTPError

api_url = "https://api.theauthapi.com/api-keys"
access_key = "REPLACE_WITH_YOUR_ACCESS_KEY"

headers = {
    "x-api-key": access_key,
}

def fetch_api_keys(filters=None):
    try:
        response = requests.get(api_url, headers=headers, params=filters)
        response.raise_for_status()
        return response.json()
    except HTTPError as error: 
        # handle error
        print(error)


filters = {
    'name': 'OPTIONAL_NAME_FILTER',
    'customUserId': 'OPTIONAL_CUSTOM_USER_ID_FILTER',
    'customAccountId': 'OPTIONAL_CUSTOM_ACCOUN_ID_FILTER',
    'isActive': "OPTIONAL_IS_ACTIVE_FILTER"
}

# fetch all api keys
api_key = fetch_api_keys()

# fetch all inactive (revoked) api keys
revoked_api_keys = fetch_api_keys({
    'isActive': False
})
print(revoked_api_keys)

# fetch api keys where customUserId equals "C11"
api_keys_filtered_by_customAccountId = fetch_api_keys({
    'customAccountId': 'C11'
})
print(api_keys_filtered_by_customAccountId)
